### DISCUSSION TITLE
Markdown inside <p></p> is not parsed

### DISCUSSION THREAD

**Discussion by vedmant** [markedjs/marked#1723] [Ideas]
**Describe the bug**

Markdown inside `<p></p>` is not parsed

**To Reproduce**

1. [Marked Demo](https://marked.js.org/demo/?text=%3Cp%3Efdsfsd%20**test**%20sdfsdfs%3C%2Fp%3E&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)
2. [CommonMark Demo](https://spec.commonmark.org/dingus/?text=%3Cp%3Efdsfsd%20**test**%20sdfsdfs%3C%2Fp%3E%0A)

**Expected behavior**

Should parse markdown inside html `<p>` tag.

**Comment by UziTech** [comment]
This follows the CommonMark [spec](https://spec.commonmark.org/0.29/#html-blocks):

> An HTML block is a group of lines that is treated as raw HTML (and will not be escaped in HTML output).

If you put a blank like between the `<p>` tags and the content it will be parsed as markdown.

[demo](https://marked.js.org/demo/?text=%3Cp%3E%0A%0Afdsfsd%20**test**%20sdfsdfs%0A%0A%3C%2Fp%3E&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)
```html
<p>

fdsfsd **test** sdfsdfs

</p>
```

Result:

<p>

fdsfsd **test** sdfsdfs

</p>

Or you can continue using HTML

```html
<p>fdsfsd <strong>test</strong> sdfsdfs</p>
```

**Comment by vedmant** [comment]
@UziTech I see, thanks. Would be nice to have an option to force parse markdown inside html tags. It will solve lots of issues with markdown in WYSIWYG editors.

**Comment by UziTech** [comment]
You could create a [renderer](https://marked.js.org/#/USING_PRO.md#renderer) for that.

Something like:

```js
const marked = require('marked');

// Override html renderer
const renderer = {
  html(html) {
    // pull out inner markdown
    const match = html.match(/^(<[^>]+>)(.*)(<[^>]+>)$/);
    // run markdown through marked again
    return match[1] + marked(match[2]) + match[3];
  }
};

marked.use({ renderer });

console.log(marked('<p>asdf **test** asdf</p>'));
// <p><p>asdf <strong>test</strong> asdf</p></p>
```

**Comment by neopostmodern** [comment]
@UziTech I'm trying to create a render like you mentioned, but can't figure it out. Consider the following input: 
```markdown
<div>
  <h1>Content</h1>
  like, *markdown*;  
  or, [links](//example.com)
</div>
```

The first iteration of your modified renderer runs fine and strips away the `<div>` and `</div>`. But the remaining content
```markdown
  <h1>Content</h1>
  like, *markdown*;  
  or, [links](//example.com)
  
```
gets passed into the HTML renderer override — and can't be matched since it's a mix of HTML and markdown and there is no wrapping HTML tag. I would have thought the tokenizer takes care of this?

Any hints?

The last version where HTML can be mixed seamlessly with markdown seems to be 0.3.1, so I'll have to downgrade for now.

**Comment by UziTech** [comment]
@neopostmodern my code above is just a proof of concept, you will have to change it to work with your use case. There are a lot of edge cases that need to be thought of to make it work with any markdown.
